### ISSUE TITLE
Cannot link libzmq.a on Android Studio (unsupported dynamic reloc R_ARM_REL32)

### ISSUE THREAD

**Issue by vb-mich** [zeromq/libzmq#3881]
# Issue description
I successfully built on Linux libzmq for Android following the instructions [here](https://github.com/zeromq/libzmq/blob/master/builds/android/README.md), now I'm trying to compile (for Android) a shared library and link to it a static build of libzmq. 
The linker gives me this errors:
`libzmq/armeabi-v7a/libzmq.a(src_libzmq_la-address.o): requires unsupported dynamic reloc R_ARM_REL32; recompile with -fPIC`
`libzmq/armeabi-v7a/libzmq.a(src_libzmq_la-ctx.o): requires unsupported dynamic reloc R_ARM_REL32; recompile with -fPIC`

I confirm the static library has been built with fPIC as well as my library
I also tried the solution here https://github.com/zeromq/libzmq/issues/2300 with no luck.

Digging on google I found this https://github.com/gavinpugh/vs-android/issues/124
Maybe a bug of Android Studio?

# Environment
libzmq built on Ubuntu Linux
Android Studio 3.6.3 running on Windows 10
NDK r21 (on both win and linux)
* libzmq version (commit hash if unreleased):  latest

**Comment by austinbenincasa** [comment]
I have experienced this issue as well on Ubuntu 16.04 gcc-8 c++17 libzmq-4.3.2

**Comment by vb-mich** [comment]
> I have experienced this issue as well on Ubuntu 16.04 gcc-8 c++17 libzmq-4.3.2

Are you using Android Studio as well?

**Comment by austinbenincasa** [comment]
No, forgot to follow up I had to build libzmq using the following, `./autogen.sh --with-pic` then `./configure`
